internal/hub/hubclient/hubclient.go

```diff
@@ -109,6 +111,58 @@ func PostClusterInfoToHub(ctx context.Context, spokeclient client.Client, hubCli
 	return nil
 }
 
+func PostClusterCredsToHub(ctx context.Context, spokeclient client.Client, hubClient client.Client, secret *corev1.Secret) error {
+	secretName := os.Getenv("CLUSTER_NAME") + "-kubernetes-dashboard"
+	err := createorUpdateClusterSecretOnHub(ctx, secretName, secret, hubClient)
+	if err != nil {
+		log.Error(err, "Error creating secret on hub cluster")
+		return err
+	}
+	return updateClusterCredsToHub(ctx, spokeclient, hubClient, secretName)
+
+}
+
+func createorUpdateClusterSecretOnHub(ctx context.Context, secretName string, secret *corev1.Secret, hubClient client.Client) error {
+	if secret.Data == nil {
+		return errors.New("secret data is nil")
+	}
+	secretData := map[string][]byte{
+		"token":  secret.Data["token"],
```

Comment:
maybe adding a check for value's existence can help, a log message saying "token" is missing or something ?